src/malli/core.cljc

```diff
@@ -2387,6 +2387,9 @@
      (let [name' `'~(symbol (str name))
            ns' `'~(symbol (str *ns*))
            sym `'~(symbol (str *ns*) (str name))]
+       ;; in cljs we need to register the schema in clojure (the cljs compiler) so it is visible in the -function-schemas* map at macroexpansion time.
+       (when (some? (:ns &env))
+         (-register-function-schema! (symbol (str *ns*)) name value (meta name)))
```

Comment:
Because of this change, this doesn't work now:

```clj
(def small-int [:int {:max 6}])

(defn plus [x] (inc x))
(m/=> plus [:=> [:cat :int] small-int])
```

... as the `small-int` is not resolved at macro-expansion time and the `small-int` is not resolved yet.

any ideas?